pipelines/rj_crm__relatorio_salesforce/tasks.py

```
@@ -0,0 +1,405 @@
+  CAST(COALESCE(SUM(CASE WHEN m.ssot__AiAgentInteractionMessageType__c = 'Input' THEN 1 ELSE 0 END), 0) AS INTEGER),
+  CAST(COALESCE(SUM(CASE WHEN m.ssot__AiAgentInteractionMessageType__c = 'Output' THEN 1 ELSE 0 END), 0) AS INTEGER)
+FROM ssot__AiAgentSession__dlm sess
+LEFT JOIN ssot__AiAgentInteraction__dlm i
+  ON i.ssot__AiAgentSessionId__c = sess.ssot__Id__c
+LEFT JOIN ssot__AiAgentInteractionMessage__dlm m
+  ON m.ssot__AiAgentInteractionId__c = i.ssot__Id__c
+WHERE sess.ssot__RelatedMessagingSessionId__c IS NOT NULL
+  AND sess.ssot__StartTimestamp__c >= '{start_date}'
+  AND sess.ssot__StartTimestamp__c < '{end_date_exclusive}'
+GROUP BY sess.ssot__Id__c, sess.ssot__RelatedMessagingSessionId__c
+ORDER BY MAX(sess.ssot__StartTimestamp__c) DESC"""
```

Ruff (0.16.0)

[error] Possible SQL injection vector through string-based query construction

(S608)